packages/elements/src/components/file-picker/file-picker.component.ts

```diff
@@ -0,0 +1,279 @@
+import { html, LitElement, nothing } from 'lit';
+import { map } from 'lit/directives/map.js';
+import { customElement, property, state } from 'lit/decorators.js';
+import { tailwind } from '../../style/index.js';
+import type { FileUnderTestModel, Metrics, MetricsResult, MutationTestMetricsResult, TestFileModel, TestMetrics } from 'mutation-testing-metrics';
+import { classMap } from 'lit/directives/class-map.js';
+import { renderIf } from '../../lib/html-helpers.js';
+
+@customElement('mte-file-picker')
+export class MutationTestReportFilePickerComponent extends LitElement {
+  static styles = [tailwind];
+
+  #currentPressedKeys = new Set<string>();
+  #searchMap = new Map<string, FileUnderTestModel | TestFileModel>();
+
+  @property({ type: Object })
+  public declare rootModel: MutationTestMetricsResult;
+
+  @state()
+  public declare openPicker: boolean;
+
+  @state()
+  public declare filteredFiles: { name: string; file: FileUnderTestModel | TestFileModel }[];
+
+  @state()
+  public declare fileIndex: number;
+
+  constructor() {
+    super();
+
+    this.openPicker = false;
+    this.fileIndex = 0;
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+
+    this.#prepareMap();
+    this.#filter('');
+
+    document.addEventListener('keydown', (e) => this.#handleKeyDown(e));
+    document.addEventListener('keyup', (e) => this.#handleKeyUp(e));
+  }
+
+  open() {
+    this.openPicker = true;
+  }
+
+  render() {
+    if (!this.openPicker) {
+      return nothing;
+    }
+
+    return html`
+      <div
+        id="backdrop"
+        @click="${() => this.#closePicker()}"
+        class="fixed left-0 top-0 z-50 flex h-full w-full justify-center bg-black/50 backdrop-blur-lg"
+      >
+        <div
+          @click="${(e: MouseEvent) => e.stopPropagation()}"
+          role="dialog"
+          id="picker"
+          class="m-4 flex h-fit h-full max-h-[500px] w-full max-w-[40rem] flex-col rounded-lg bg-gray-200/60 p-4 backdrop-blur-lg md:w-1/2"
+        >
+          <div class="mb-2 flex items-center rounded bg-gray-200/60 p-2 backdrop-blur-lg">
+            <div class="mx-2 flex items-center text-gray-800">
+              <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="size-6">
+                <path
+                  stroke-linecap="round"
+                  stroke-linejoin="round"
+                  d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
+                />
+              </svg>
+            </div>
+            <input
```

Comment:
Input components should always have a `<label>` to describe what the input value does, for accessibility. MDN has a very good page about it if you need some examples: https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML